pkg/cloudprovider/providers/gce/gce_loadbalancer_wrapper.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+)
+
+type wrappedLoadBalancer struct {
+	gce *GCECloud
+}
+
+func newWrappedLoadBalancer(gce *GCECloud) cloudprovider.LoadBalancer {
+	return &wrappedLoadBalancer{gce: gce}
+}
+
+func (w *wrappedLoadBalancer) GetLoadBalancer(clusterName string, service *v1.Service) (status *v1.LoadBalancerStatus, exists bool, err error) {
+	svc := w.convertServiceLBAnnotations(service)
+	return w.gce.GetLoadBalancer(clusterName, svc)
+}
+
+func (w *wrappedLoadBalancer) EnsureLoadBalancer(clusterName string, service *v1.Service, nodes []*v1.Node) (*v1.LoadBalancerStatus, error) {
+	svc := w.convertServiceLBAnnotations(service)
+	if err := validateServiceLBAnnotations(svc); err != nil {
```

Comment:
We also call this func within [`gce_loadbalaner.go:EnsureLoadBalancer()`](https://github.com/yujuhong/kubernetes/blob/7fab6c8755ff99ceb3e68cdd225c579274070d1c/pkg/cloudprovider/providers/gce/gce_loadbalancer.go#L119-L122), might just keep one?

Reply:
Yep, that'd need to be removed if reviewers like this commit better. The PR pivoted with the latest commit, so I just wanted to get feedback first on the overall approach. I'll make it more clear in the PR description :-)